### PR TITLE
*: closed immediately if error rate is changed to 0

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #8678.

### What is changed and how does it work?

This PR does the following changes:
1. add more tests for the circuit breaker.
2. change to closed state immediately if the error rate is changed to 0

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by okJiang — COMMENTED** [review]
LGTM, cc @Tema

**Review by nolouch — APPROVED** [review]

File: client/circuitbreaker/circuit_breaker.go [hunk 1]
```diff
@@ -19,13 +19,11 @@ import (
 	"sync"
 	"time"
 
-	"github.com/tikv/pd/client/errs"
-
+	"github.com/pingcap/log"
```
[the remark this answers is not in the document]
Comment (rleungx) [reply client/circuitbreaker/circuit_breaker.go:22]: I add a linter for it, see https://github.com/tikv/pd/pull/8925

File: client/circuitbreaker/circuit_breaker.go [hunk 2]
```diff
@@ -253,29 +252,37 @@ func (s *State[T]) onRequest(cb *CircuitBreaker[T]) (*State[T], error) {
...
+				zap.Any("config", cb.config))
 			return cb.newState(now, StateHalfOpen), nil
 		} else {
 			// continue in the open state till CoolDownInterval is over
 			return s, errs.ErrCircuitBreakerOpen
 		}
 	case StateHalfOpen:
+		if s.cb.config.ErrorRateThresholdPct == 0 {
+			return cb.newState(now, StateClosed), nil
+		}
```
[the remark this answers is not in the document]
Comment (rleungx) [reply client/circuitbreaker/circuit_breaker.go:272]: added, PTAL

**Review by okJiang — APPROVED** [review]

**Pull request merged**